src/common/http/HttpClient.h

```diff
@@ -1,47 +1,63 @@
-/* Copyright (c) 2019 vesoft inc. All rights reserved.
+/* Copyright (c) 2022 vesoft inc. All rights reserved.
  *
  * This source code is licensed under Apache 2.0 License.
  */
 
-#ifndef COMMON_HTTPCLIENT_H
-#define COMMON_HTTPCLIENT_H
+#ifndef COMMON_HTTP_HTTPCLIENT_H_
+#define COMMON_HTTP_HTTPCLIENT_H_
+#include <memory>
+#include <mutex>
+#include <string>
 
-#include "common/base/Base.h"
 #include "common/base/StatusOr.h"
+#include "curl/curl.h"
 
 namespace nebula {
-namespace http {
+
+class CurlHandle {
+ public:
+  static CurlHandle* instance();
+
+ private:
+  CurlHandle();
+  ~CurlHandle();
+};
+
+struct HttpResponse {
+  CURLcode curlCode;
+  std::string curlMessage;
+  std::string header;
+  std::string body;
+};
 
 class HttpClient {
  public:
-  HttpClient() = delete;
-
-  ~HttpClient() = default;
-  // Send a http GET request
-  static StatusOr<std::string> get(const std::string& path, const std::string& options = "-G");
-
-  // Send a http POST request
-  static StatusOr<std::string> post(const std::string& path, const std::string& header);
-  // Send a http POST request with a different function signature
-  static StatusOr<std::string> post(const std::string& path,
-                                    const std::unordered_map<std::string, std::string>& header);
-  static StatusOr<std::string> post(const std::string& path,
-                                    const folly::dynamic& data = folly::dynamic::object());
-
-  // Send a http PUT request
-  static StatusOr<std::string> put(const std::string& path, const std::string& header);
-  static StatusOr<std::string> put(const std::string& path,
-                                   const std::unordered_map<std::string, std::string>& header);
-  static StatusOr<std::string> put(const std::string& path,
-                                   const folly::dynamic& data = folly::dynamic::object());
-
- protected:
-  static StatusOr<std::string> sendRequest(const std::string& path,
-                                           const folly::dynamic& data,
-                                           const std::string& reqType);
+  static HttpResponse get(const std::string& url);
+  static HttpResponse get(const std::string& url, const std::vector<std::string>& headers);
+
+  static HttpResponse post(const std::string& url,
+                           const std::vector<std::string>& headers,
+                           const std::string& body);
+  static HttpResponse delete_(const std::string& url, const std::vector<std::string>& headers);
```

Comment:
`del`? Not a big issue, up to you

Reply:
Emm, both are OK. But I don't want to change it:)
Maybe someone who doesn't like delete_ will modify it one day.